src/transformStateWithClones.js

```diff
@@ -7,7 +7,43 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const result = [];
+  const workState = { ...state };
+
+  for (const i of actions) {
+    for (const key in i) {
+      const add = i.extraData;
+      const remove = i.keysToRemove;
+
+      if (i[key] === 'addProperties') {
+        for (const x in add) {
+          workState[x] = add[x];
+        }
+
+        result.push({ ...workState });
+      }
+
+      if (i[key] === 'removeProperties') {
+        for (const ch of remove) {
+          if (ch in workState) {
+            delete workState[ch];
+          }
+        }
+
+        result.push({ ...workState });
+      }
+
+      if (i[key] === 'clear') {
+        for (const del in workState) {
+          delete workState[del];
+        }
+
+        result.push({ ...workState });
+      }
```

Comment:
Do not repeat your self push result after all iterations
```suggestion

      }

      if (i[key] === 'removeProperties') {
        for (const ch of remove) {
          if (ch in workState) {
            delete workState[ch];
          }
        }

      }

      if (i[key] === 'clear') {
        for (const del in workState) {
          delete workState[del];
        }
     
      }
      result.push({ ...workState });
```